examples/hygiene.kl

```diff
@@ -1,23 +1,64 @@
 #lang "prelude.kl"
 
-[import [shift "prelude.kl" 1]]
-[import [shift "quasiquote.kl" 1]]
+(import (shift "prelude.kl" 1))
+(import (shift "quasiquote.kl" 1))
+(import "define-syntax-rule.kl")
 
-[define fun [lambda (x y) x]]
 
-[define-macros ([should-not-be-captured [lambda (stx) [pure [quote fun]]]])]
+-- Klister's macro system is "hygienic", meaning that it automatically prevents
+-- accidental variable capture. For example, in the code below, variables
+-- defined at the macro's call site are not accidentally captured by the
+-- variables uses in the code generated by the "should-not-be-captured" macro.
 
-[define fun [lambda (x y) y]]
+(define first (lambda (x y) x))
+(define second (lambda (x y) y))
 
-[example (fun 'a 'b)]
-[example ((should-not-be-captured) 'a 'b)]
+(define fun first)
 
-[define-macros
-  ([should-not-capture
-    [lambda (stx)
-      (syntax-case stx
-        [[list [_ body]]
-         [pure `(lambda (fun) ,body)]]
-        [_ (syntax-error '"bad syntax" stx)])]])]
+(define-syntax-rule (should-not-be-captured)
+  fun)
 
-[example ([should-not-capture fun] [lambda (x y) x] 'a 'b)]
+(define fun second)
+
+(example (fun 'a 'b))                       -- (1)
+(example ((should-not-be-captured) 'a 'b))  -- (2)
+
+-- The function "fun" is defined twice: first as a function which returns its
+-- first argument, and then as a function which returns its second argument.
+-- The second definition shadows the first, thus on line (1), "fun" returns its
+-- second argument.
+--
+-- Between the two definition of "fun", a macro "should-not-be-captured" is
+-- defined. It is a very simple macro which always expands to the symbol 'fun.
+-- Thus, line (2) expands to (fun 'a 'b), same as line (1). And yet, thanks to
+-- hygiene, this time the symbol "fun" refers to the first definition, the one
+-- which was in scope when the macro was defined! This allows the user to
+-- redefine a common identifier such as "case" or "lambda" without accidentally
+-- changing the semantics of all the macros macros which produce code
+-- containing those symbols.
```

Comment:
Here and at the very end of the file, I hint at more practical situations in which hygiene is useful. Perhaps I should just use those practical situations directly as the example code?

The reason I hesitate to make this change is that the existing examples are elegantly minimal. I don't know if they are minimal because Klister was so hard to use back then that being minimal was necessary in order to keep the code short, for pedagogical reasons, or because these are standard examples in the hygiene litterature.

Reply:
Never mind, I have changed my mind about this PR, I no longer want to explain why hygiene is useful.